godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/ClassHandle.kt

```diff
@@ -0,0 +1,86 @@
+package godot.core
+
+import godot.gdnative.*
+import godot.registration.RPCMode
+import kotlinx.cinterop.*
+
+internal class ClassHandle<T: Object>(
+    private val nativescriptHandle: COpaquePointer,
+    private val className: String,
+    private val parentClassName: String,
+    private val factory: () -> T,
+    private val isTool: Boolean
+) {
+    private val disposables = mutableListOf<COpaquePointer>()
+
+    fun wrap(instance: COpaquePointer): T {
+        return Godot.noInitZone {
+            factory().also { it.ptr = instance }
+        }
+    }
+
+    fun init() {
+        memScoped {
+            val methodData = StableRef.create(this@ClassHandle).asCPointer()
+            // register constructor and destructor
+            val create = cValue<godot_instance_create_func> {
+                create_func = staticCFunction(::createInstance)
+                free_func = staticCFunction(::disposeClassHandle)
+                method_data = methodData
+            }
+            val destroy = cValue<godot_instance_destroy_func> {
+                destroy_func = staticCFunction(::destroyInstance)
+                method_data = methodData
+            }
+            val registerMethod = if (isTool) {
+                Godot.nativescript.godot_nativescript_register_tool_class
```

Comment:
Maybe we can avoid to call nativescript api wrapper when calling the classic api. So that we avoid overhead even if it is not that much.

Reply:
@piiertho sorry, what do you mean?